compiler/qsc_frontend/src/resolve.rs

```diff
@@ -550,26 +550,31 @@ impl GlobalTable {
         package: &ast::Package,
     ) -> Vec<Error> {
         let mut errors = Vec::new();
-        for namespace in &*package.namespaces {
-            self.names.insert(
-                namespace.name.id,
-                Res::Item(intrapackage(assigner.next_item())),
-            );
-            self.scope
-                .namespaces
-                .insert(Rc::clone(&namespace.name.name));
-
-            for item in &*namespace.items {
-                match bind_global_item(
-                    &mut self.names,
-                    &mut self.scope,
-                    &namespace.name.name,
-                    || intrapackage(assigner.next_item()),
-                    item,
-                ) {
-                    Ok(()) => {}
-                    Err(error) => errors.push(error),
+        for node in &*package.nodes {
+            match node {
+                TopLevelNode::Namespace(namespace) => {
+                    self.names.insert(
+                        namespace.name.id,
+                        Res::Item(intrapackage(assigner.next_item())),
+                    );
+                    self.scope
+                        .namespaces
+                        .insert(Rc::clone(&namespace.name.name));
+
+                    for item in &*namespace.items {
+                        match bind_global_item(
+                            &mut self.names,
+                            &mut self.scope,
+                            &namespace.name.name,
+                            || intrapackage(assigner.next_item()),
+                            item,
+                        ) {
+                            Ok(()) => {}
+                            Err(error) => errors.push(error),
+                        }
+                    }
                 }
+                TopLevelNode::Stmt(_) => todo!("Not yet supported for top-level statements"),
```

Comment:
nitpick: I think we'd want to make this `unimplemented!()` instead of `todo!()` because right now we have no reason to believe we will come back to support this (top-level statements aren't something we add here, they are handled during incremental compilation). There is a subtle distinction between the two macros as called out by the docs at https://doc.rust-lang.org/std/macro.unimplemented.html:

> The difference between unimplemented! and todo! is that while todo! conveys an intent of implementing the functionality later and the message is “not yet implemented”, unimplemented! makes no such claims. Its message is “not implemented”. Also some IDEs will mark todo!s.

Reply:
I didn't know about the difference, so thanks for that! But I also just realized I don't quite remember what I was going for with this change. It looks unfinished, honestly. This method is only called for batch compilation, and it's suspiciously similar to `bind_namespace_items` which is called for the incremental path. I was trying to consolidate these two paths but I think I just missed this one. I'll take a closer look to see if I can clean it up further.